syntax = "proto3";
package quest.v1;

import "google/protobuf/struct.proto";
import "common.proto";

// QuestObjective: Single objective in a quest
message QuestObjective {
  string objective_id = 1;
  string description = 2;
  string type = 3;
  map<string, string> params = 4;
}

// Quest: Complete quest structure
message Quest {
  string quest_id = 1;
  string title = 2;
  string description = 3;
  repeated QuestObjective objectives = 4;
  string difficulty = 5;
  map<string, string> tags = 6;
}

// QuestGenerationRequest: Request to generate a quest
message QuestGenerationRequest {
  common.v1.Meta meta = 1;
  string player_id = 2;
  uint64 seed = 3;
  google.protobuf.Struct context = 4;
  string difficulty = 5; // EASY, NORMAL, HARD
}

// QuestGenerationResponse: Response with generated quest
message QuestGenerationResponse {
  common.v1.Meta meta = 1;
  Quest quest = 2;
  common.v1.Error error = 3;
}

